MaxText/layers/linears.py

```diff
@@ -429,37 +436,64 @@ def generate_masks(self, top_k_indices):
       expert_mask = jax.nn.one_hot(top_k_indices, num_classes=self.num_experts, dtype=jnp.int32)
       reshaped_expert_mask = jnp.reshape(expert_mask, (batch_size, seq_len * self.num_experts_per_tok, self.num_experts))
       expert_token_count = jnp.cumsum(reshaped_expert_mask, axis=1)
-      expert_token_count = jnp.reshape(expert_token_count, ((batch_size, seq_len, self.num_experts_per_tok, self.num_experts)))
-      trunc_expert_mask = expert_mask * jnp.less_equal(expert_token_count, expert_capacity)
+      reshaped_expert_token_count = jnp.reshape(expert_token_count, ((batch_size, seq_len, self.num_experts_per_tok, self.num_experts)))
```

Comment:
would be helpful to have consistent naming - above we have `reshaped_expert_mask` which has seq_len * experts_per_tok, but here we have `reshaped_expert_token_count, which is the opposite (seq_len, experts_per_tok). Maybe expert_token_count_seq_fused, where fused refers to seq_len * experts_per_tok (incluce comment for first *fused variable name)

Reply:
Thanks! Good point! Let me update those variables and try to keep them consistent.

Reply:
Updated variables with `fused` suffix if the shape like (batch, seq * num_experts_per_tok, ...).